File: atosim/org.palladiosimulator.blockchainsystems.threesim/src/main/java/org/palladiosimulator/blockchainsystems/threesim/metrics/ForkProbability.java
package org.palladiosimulator.blockchainsystems.threesim.metrics;
import org.palladiosimulator.blockchainsystems.threesim.metrics.abstractions.OutputMetric;
public class ForkProbability implements OutputMetric<Double> {
    public static final String NAME = "ForkProbability";
    public static final String UNIT = null;
    private final double _value;
    public ForkProbability(double value) { _value = value; }
    @Override public Double getValue() { return _value; }
    @Override public String getName() { return NAME; }
    @Override public String getUnit() { return UNIT; }
}

File: atosim/org.palladiosimulator.blockchainsystems.threesim/src/main/kotlin/org/palladiosimulator/blockchainsystems/threesim/metrics/ForkProbability.kt
package org.palladiosimulator.blockchainsystems.threesim.metrics

import kotlinx.serialization.Serializable
import org.palladiosimulator.blockchainsystems.threesim.metrics.abstractions.OutputMetric

@Serializable
class ForkProbability(
    override val value: Double
) : OutputMetric<Double> {

    companion object {
        const val NAME = "ForkProbability"
        val UNIT:String? = null
    }

    override val name: String = NAME
    override val unit: String? = UNIT
}